### ISSUE TITLE
feature: Create Riot API integration

### ISSUE THREAD

**Issue by zyzmoz** [zyzmoz/global-player#9]
# Overview

Create a function to call [/lol/league/v4/challengerleagues/by-queue/{queue}](https://developer.riotgames.com/apis#league-v4/GET_getChallengerLeague)

# Acceptance Criteria

The call should get `challenger` players from the `NA` server

# Out of Scope

At this moment we're not saving data to the database

**NOTES:** We already have an API KEY

Blocked by #96 #98 #95 #97

**Comment by zyzmoz** [comment]
# Data Model

- LeagueV4: Players
  
  - AccountV4: puuid
    
  - MatchesV4: Matches
    
  - MatchesV4: Match details
    

Get Challenger Players:

[/lol/league/v4/challengerleagues/by-queue/{queue}](https://developer.riotgames.com/apis#league-v4/GET_getChallengerLeague)

Get `puuid` -> `playerId` / `profileIconId` -> `playerImage`:

[/lol/summoner/v4/summoners/{encryptedSummonerId}](https://developer.riotgames.com/apis#summoner-v4/GET_getBySummonerId)

Get Match Ids: 
[/lol/match/v5/matches/by-puuid/{puuid}/ids](https://developer.riotgames.com/apis#match-v5/GET_getMatchIdsByPUUID)

Get Match Details:

[/lol/match/v5/matches/{matchId}](https://developer.riotgames.com/apis#match-v5/GET_getMatch)

Maximum Players: 2400

20 requests every 1 seconds  
100 requests every 2 minutes

## Data needed

MatchesV4 -> `teamPosition`, `kills`, `deaths`, `assists`, `championName`, `championId`, `win`

Champion Image http://ddragon.leagueoflegends.com/cdn/12.19.1/img/champion/Aatrox.png

Summoner Icon

https://ddragon.leagueoflegends.com/cdn/12.20.1/img/profileicon/27.png

| Data | Source | Property | Type |
| --- | --- | --- | --- |
| Rank |     |     |     |
| Role | MatchesV4 | `teamPosition` | string |
| Champion | MatchesV4 | `championName`, `championId` | string, number |
| Win/Loss | MatchesV4 | `win` | boolean |
| Kills | MatchesV4 | `kills` | number |
| Deaths | MatchesV4 | `deaths` | number |
| Assists | MatchesV4 | `assists` | number |
